### PR TITLE
[IBU-10659] update report

### PR DESCRIPTION
**Pull request by laisecf**
Update report: add number of Swiss-Prot proteins and example of Swiss-Prot accession (one Swiss-Prot accession per lost or gained Swiss-Prot description) [Jira Ticket](https://www.ebi.ac.uk/panda/jira/browse/IBU-10659)

### CODE REVIEW

**Review by matthiasblum — CHANGES_REQUESTED** [review]
In good shape, thanks.

I've highlighted the few issues I identified.

Other improvements that would be nice:

1. `get_swissprot_descriptions` now returns a list of UniProt accessions for each description, you could return a _set_ of UniProt accessions so we don't need to convert the list to set when comparing descriptions before and after the update
2. once `descr2prots` is populated (mapping between a description and the proteins with this description), its type is `dict[str, set[str]]`. You could transform it to `dict[str, list[str]]` so we don't need to convert the `set` of accessions to a `list` of accessions for each description, like here: https://github.com/ProteinsWebTeam/pyinterprod/blob/07fa5c67a5225d63880c476c6c956e6e55a57a65/pyinterprod/interpro/report.py#L174-L179

File: pyinterprod/interpro/report.py [hunk 1]
```diff
@@ -323,33 +347,51 @@ def send_prot_update_report(ora_url: str, pg_url: str, data_dir: str,
     cur.execute("SELECT VERSION FROM INTERPRO.DB_VERSION WHERE DBCODE = 'u'")
     release, = cur.fetchone()
 
+    desc2prot = {}
     # Load entry -> descriptions BEFORE UniProt update
```
[the remark this answers is not in the document]
Comment (laisecf) [reply pyinterprod/interpro/report.py:351]: bur desc2prot is used in descriptions "before" and "after" UniProt update

File: pyinterprod/interpro/report.py [hunk 2]
```diff
@@ -113,26 +113,42 @@ def send_db_update_report(ora_url: str, pg_url: str, dbs: list[Database],
...
+                    desc2prot[descr] = set(proteins)
 
-            try:
-                entry_acc, entry_type, entry_name, _ = integrated[acc]
-            except KeyError:
-                continue
+                try:
+                    entry_acc, entry_type, entry_name, _ = integrated[acc]
+                except KeyError:
+                    continue
```
[the remark this answers is not in the document]
Comment (laisecf) [reply pyinterprod/interpro/report.py:135]: you right! this block dont need to be in the internal for

File: pyinterprod/interpro/report.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthiasblum) [reply pyinterprod/interpro/report.py:351]: Fair enough.

File: pyinterprod/interpro/report.py [hunk 3]
```diff
@@ -150,14 +166,22 @@ def send_db_update_report(ora_url: str, pg_url: str, dbs: list[Database],
...
             link = f"{pronto_link}/signatures/{acc}/descriptions/?reviewed"
+
+            lost_descs = [
+                f"{desc} ({list(descr2prots[desc])[0]})" for desc in sorted(lost)
+            ]
+            gained_descs = [
+                f"{desc} ({list(descr2prots[desc])[0]})" for desc in sorted(gained)
+            ]
             fh.write(f"{acc}\t{link}\t{entry_acc}\t{types[entry_type]}"
+                     f"\t{len(sig2prots[acc])}\t"
```
Comment (laisecf) [inline pyinterprod/interpro/report.py:181]:
```suggestion
                     f"\t{len(sig2prots[acc])}"
```

**Review by matthiasblum — CHANGES_REQUESTED** [review]

File: pyinterprod/interpro/report.py [hunk 4]
```diff
@@ -150,14 +166,22 @@ def send_db_update_report(ora_url: str, pg_url: str, dbs: list[Database],
...
                      f"\t{entry_name}\t{len(lost)}\t{len(gained)}"
-                     f"\t{' | '.join(sorted(lost))}"
-                     f"\t{' | '.join(sorted(gained))}\n")
+                     f"\t{' | '.join(lost_descs)}"
+                     f"\t{' | '.join(gained_descs)}\n")
```
[the remark this answers is not in the document]
Comment (laisecf) [reply pyinterprod/interpro/report.py]: You're right! I had fixed this and then forgot about it and came back to it when I refactored... sorry for this o/

**Review by matthiasblum — APPROVED** [review]
Looks good! :+1: 
Feel free to merge. :rocket:

**Pull request merged**